### PR TITLE
Alter BEEFY primitives to prepare for potential BLS integration

### PR DESCRIPTION
**Pull request by tomusdrw**
This PR makes `SignedCommitment` contain a generic parameter `Signature` (instead of harcoded `ecsda` signature), so that in the future it can be used as: `SignedCommitment<BlockNumber, ecsda::Signature>` or `SignedCommitment<BlockNumber, (ecdsa::Signature, (bls::Signature, ecsda::Signature))>`).

Also I've renamed `VersionedCommitment` to `VersionedFinalityProof`. Currently BEEFY finality proof is simply the `SignedCommitment` with sufficient number of valid signatures, but in the future it will include not only the `ecdsa` Signatures, but also an aggregated BLS signature.

Related: https://github.com/paritytech/grandpa-bridge-gadget/issues/2

CC @drskalman @andresilva

### CODE REVIEW

File: primitives/beefy/src/commitment.rs [hunk 1]
```diff
@@ -183,28 +183,28 @@ struct CompactSignedCommitment<TCommitment> {
...
 	/// efficient network transport.
-	fn pack(signed_commitment: &'a SignedCommitment<TBlockNumber>) -> Self {
+	fn pack(signed_commitment: &'a SignedCommitment<TBlockNumber, TSignature>) -> Self {
 		let SignedCommitment { commitment, signatures } = signed_commitment;
 		let validator_set_len = signatures.len() as u32;
 		let mut signatures_from: BitField = vec![];
-		let mut signatures_compact: Vec<Signature> = vec![];
+		let mut signatures_compact: Vec<&'a TSignature> = vec![];
+
+		let mut bits: Vec<u8> =
```
Comment (kianenigma) [inline primitives/beefy/src/commitment.rs:198]: can't we do this and the next block in one iteration?

**Review by kianenigma — COMMENTED** [review]
Not my thing, but the changes make sense to me.

File: primitives/beefy/src/commitment.rs (hunk 1 above)
Comment (tomusdrw) [reply primitives/beefy/src/commitment.rs:198]: Yes, but actually I think I'd rather use iterators for both:
```rust
let mut bits = signatures.iter().map(|x| if x.is_some() { 1 } else { 0 }).collect();
let mut signatures_compact = signatures.iter().filter_map(|x| x).collect();
```

wdyt?

**Review by acatangiu — APPROVED** [review]
Thanks!

**Review by svyatonik — APPROVED** [review]

**Comment by drskalman** [comment]
@andresilva, @acatangiu: The current structure results in BEEFY messages of around 80K for 1K validators vs if we just keep one aggregated BLS signature it would 32K + 48Bytes + 1000/8  = 32172 that is about 2.5 time bulkier messages for no obvious advantage.

I suggest to have a vector of ECDSA signatures and a  single aggregated BLS signature field in the BEEFY message instead of having a vector of (ECDSA, single BLS) signatures.

**Comment by acatangiu** [comment]
> I suggest to have a vector of ECDSA signatures and a single aggregated BLS signature field in the BEEFY message instead of having a vector of (ECDSA, single BLS) signatures.

Agreed! I believe the ideal end result is to have single aggregated signature, resulting in slim messages.

**Comment by drskalman** [comment]
So aggregation before gossip unfortunately is not possible with unorganized gossip that we have: https://github.com/w3f/bls/issues/51

**Pull request merged**